src/vs/workbench/browser/parts/titlebar/menubarControl.ts

```diff
@@ -422,87 +423,16 @@ export class CustomMenubarControl extends MenubarControl {
-			if (activityBarInactiveFgColor) {
-				collector.addRule(`
-				.monaco-workbench .activitybar .menubar.compact > .menubar-menu-button,
-				.monaco-workbench .activitybar .menubar.compact .toolbar-toggle-more {
-					color: ${activityBarInactiveFgColor};
-				}
-				`);
-			}
-
-			const activityBarFgColor = theme.getColor(ACTIVITY_BAR_FOREGROUND);
-			if (activityBarFgColor) {
-				collector.addRule(`
-				.monaco-workbench .activitybar .menubar.compact > .menubar-menu-button.open,
-				.monaco-workbench .activitybar .menubar.compact > .menubar-menu-button:focus,
-				.monaco-workbench .activitybar .menubar.compact:not(:focus-within) > .menubar-menu-button:hover,
-				.monaco-workbench .activitybar .menubar.compact  > .menubar-menu-button.open .toolbar-toggle-more,
-				.monaco-workbench .activitybar .menubar.compact > .menubar-menu-button:focus .toolbar-toggle-more,
-				.monaco-workbench .activitybar .menubar.compact:not(:focus-within) > .menubar-menu-button:hover .toolbar-toggle-more {
-					color: ${activityBarFgColor};
-				}
-			`);
-			}
-
-			const menubarInactiveWindowFgColor = theme.getColor(TITLE_BAR_INACTIVE_FOREGROUND);
-			if (menubarInactiveWindowFgColor) {
-				collector.addRule(`
-					.monaco-workbench .menubar.inactive:not(.compact) > .menubar-menu-button,
-					.monaco-workbench .menubar.inactive:not(.compact) > .menubar-menu-button .toolbar-toggle-more  {
-						color: ${menubarInactiveWindowFgColor};
-					}
-				`);
-			}
-
-			const menubarSelectedFgColor = theme.getColor(MENUBAR_SELECTION_FOREGROUND);
-			if (menubarSelectedFgColor) {
-				collector.addRule(`
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button.open,
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button:focus,
-					.monaco-workbench .menubar:not(:focus-within):not(.compact) > .menubar-menu-button:hover,
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button.open .toolbar-toggle-more,
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button:focus .toolbar-toggle-more,
-					.monaco-workbench .menubar:not(:focus-within):not(.compact) > .menubar-menu-button:hover .toolbar-toggle-more {
-						color: ${menubarSelectedFgColor};
-					}
-				`);
-			}
-
-			const menubarSelectedBgColor = theme.getColor(MENUBAR_SELECTION_BACKGROUND);
-			if (menubarSelectedBgColor) {
-				collector.addRule(`
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button.open .menubar-menu-title,
-					.monaco-workbench .menubar:not(.compact) > .menubar-menu-button:focus .menubar-menu-title,
-					.monaco-workbench .menubar:not(:focus-within):not(.compact) > .menubar-menu-button:hover .menubar-menu-title {
-						background-color: ${menubarSelectedBgColor};
-					}
-				`);
-			}
-
 			const menubarSelectedBorderColor = theme.getColor(MENUBAR_SELECTION_BORDER);
 			if (menubarSelectedBorderColor) {
 				collector.addRule(`
 					.monaco-workbench .menubar > .menubar-menu-button:hover .menubar-menu-title  {
```

Comment:
We can still get rid of this JS by having the color as part of the `outline` rule. We learned that when a color is `undefined` the entire rule will be ignored.

Reply:
yeah, you reminded me haha

Reply:
Thank you for letting me try it!

Reply:
thanks! I'll try it